test/neo4clj/query_builder_test.clj

```diff
@@ -110,55 +137,80 @@
+       :returns ["n1" {:ref-id "n2"} "r1"]}
+
+      "MATCH (n1:Person)-[r2:MARRIED_TO]->(n3:Person) WHERE ID(n1) = 4 AND ID(n3) = 16 MATCH (n2:Company) CREATE (n1)-[r1:EMPLOYEE]->(n2) RETURN n1, n2, r1"
+      {:lookups [{:ref-id "n1" :labels [:person] :id 4 :rels [{:ref-id "r2" :type :married-to :from "n1" :to {:ref-id "n3" :labels [:person] :id 16}}]}
+                 {:ref-id "n2" :labels [:company]}]
+       :rels [{:ref-id "r1" :type :employee :from "n1" :to {:ref-id "n2"}}]
+       :returns ["n1" {:ref-id "n2"} "r1"]}
       )))
 
 (t/deftest lookup-graph-query
-  (t/testing "Cypher to lookup a graph"
-    (t/are [expected-cypher nodes rels]
-        (= expected-cypher (sut/lookup-graph-query nodes rels))
-      "MATCH (n1:Person)-[r1:OWNER]->(n2:Person) WHERE ID(n2) = 4"
-      {"n1" {:ref-id "n1" :labels [:person]}}
-      [{:ref-id "r1" :type :owner :from "n1" :to {:ref-id "n2" :labels [:person] :id 4}}]
-
-      "MATCH ()-[r1:OWNER]->()"
-      {"n1" {:ref-id "n1" :labels [:person]}}
-      [{:ref-id "r1" :type :owner :from nil}]
-
-      "MATCH ()-[r1:OWNER]->()"
-      {"n1" {:ref-id "n1" :labels [:person]}}
-      [{:ref-id "r1" :type :owner :from nil :to {}}]
-
-      "MATCH (n1:Person) WHERE NOT ()-[:OWNER]->(n1)"
-      {"n1" {:ref-id "n1" :labels [:person]}}
-      [{:ref-id "r1" :type :owner :from nil :to "n1" :exists false}]
-
-      "MATCH (n1:Person)-[r1:OWNER]->(n2:Person) WHERE ID(n2) = 4"
-      {"n1" {:ref-id "n1" :labels [:person]}
-       "n3" {:ref-id "n3" :labels [:customer]}}
-      [{:ref-id "r1" :type :owner :from "n1" :to {:ref-id "n2" :labels [:person] :id 4}}]
-
-      "MATCH (n1:Person)-[r1:OWNER]->(n2:Person)"
-      {"n1" {:ref-id "n1" :labels [:person]}
-       "n2" {:ref-id "n2" :labels [:person]}}
-      [{:ref-id "r1" :type :owner :from "n1" :to "n2"}]
-
-      "MATCH (n1:Person)-[r1:OWNER]->(n2:Person) MATCH (n1)-[r2:EMPLOYEE]->(n3:Company) WHERE ID(n2) = 4 AND ID(n3) = 6"
-      {"n1" {:ref-id "n1" :labels [:person]}
-       "n3" {:ref-id "n3" :labels [:company]:id 6}}
-      [{:ref-id "r1" :type :owner :from "n1" :to {:ref-id "n2" :labels [:person] :id 4}}
-       {:ref-id "r2" :type :employee :from "n1" :to "n3"}]
-
-      "MATCH (n1:Person)-[r1:OWNER]->(n2:Person) MATCH (n3:Company) WHERE ID(n2) = 4 AND ID(n3) = 6 AND NOT (n1)-[:EMPLOYEE]->(n3)"
-      {"n1" {:ref-id "n1" :labels [:person]}
-       "n3" {:ref-id "n3" :labels [:company]:id 6}}
-      [{:ref-id "r1" :type :owner :from "n1" :to {:ref-id "n2" :labels [:person] :id 4}}
-       {:ref-id "r2" :type :employee :from "n1" :to "n3" :exists false}]
-
-      "MATCH (n1:Person) MATCH (n3:Company) WHERE ID(n3) = 6 AND NOT (n1)-[:EMPLOYEE]->(n3)"
-      {"n1" {:ref-id "n1" :labels [:person]}
-       "n3" {:ref-id "n3" :labels [:company]:id 6}}
-      [{:ref-id "r2" :type :employee :from "n1" :to "n3" :exists false}]
-      )))
+  (let [next-gensym (atom 0)]
+    (with-redefs [cypher/gen-ref-id (fn [] (str "G__" (swap! next-gensym inc)))]
```

Comment:
I would put the `with-redefs` inside the `are` to reset the atom for every test. When used in mulitiple test cases you can re-arrange the data sets without affecting the tests. Imagine wanting the test case using `G__2` to be placed before the one using `G__1`... that would require also touching the data.

Reply:
Ignored